swarm/storage/mru/doc.go

```diff
@@ -1,52 +1,60 @@
-// Mutable resource is an entity which allows updates to a resource
+// Package mru defines Mutable resource updates.
+// A Mutable Resource is an entity which allows updates to a resource
 // without resorting to ENS on each update.
 // The update scheme is built on swarm chunks with chunk keys following
 // a predictable, versionable pattern.
 //
-// Updates are defined to be periodic in nature, where periods are
-// expressed in terms of number of blocks.
+// Updates are defined to be periodic in nature, where the update frequency
+// is expressed in seconds.
 //
-// The root entry of a mutable resource is tied to a unique identifier,
-// typically - but not necessarily - an ens name.  The identifier must be
-// an valid IDNA string. It also contains the block number
-// when the resource update was first registered, and
-// the block frequency with which the resource will be updated, both of
+// The root entry of a mutable resource is tied to a unique identifier that
+// is deterministically generated out of the metadata content that describes
+// the resource. This metadata includes a user-defined resource name, a resource
+// start time that indicates when the resource becomes valid,
+// the frequency in seconds with which the resource is expected to be updated, both of
 // which are stored as little-endian uint64 values in the database (for a
-// total of 16 bytes). It also contains the unique identifier.
+// total of 16 bytes). It also contains the owner's address (ownerAddr)
 // This MRU info is stored in a separate content-addressed chunk
 // (call it the metadata chunk), with the following layout:
 //
-// (startblock|frequency|address|identifier)
+// (00|length|startTime|frequency|name|ownerAddr)
```

Comment:
this should just be a normal swarm chunk with content address validation.
no need for initial byte differentiator. also dont see the use of length

Reply:
It is content-addressed, but not Swarm-type. Remember the discussion about verifying resource ownership and the concept of metaHash. `rootAddr = H(ownerAddr, H(metadata))`

Again, including a header length field is always a good practice when encoding binary data, for extensibility and error checking.

The zero-byte differentiatior helps discriminate earlier at the `Validator` function, but can be removed.

Let's take the encoding discussion to #704, where it can be clearly seen in the refactored code.